core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala

```diff
@@ -259,19 +264,30 @@ import akka.stream.alpakka.cassandra.scaladsl.{ CassandraSession, CassandraSessi
     session.underlying().flatMap(_.executeAsync(batch.build()).toScala).map(_ => ())
   }
 
-  private def serialize(payload: Any): Future[Serialized] =
+  private def serialize(payload: Any, metadata: SnapshotMetadata): Future[Serialized] =
     try {
-      def serializeMeta(): Option[SerializedMeta] =
+      def serializeMeta(): Option[SerializedMeta] = {
+        def toSerializedForm(m: Any): SerializedMeta = {
+          val m2 = m.asInstanceOf[AnyRef]
+          val serializer = serialization.findSerializerFor(m2)
+          val serManifest = Serializers.manifestFor(serializer, m2)
+          val metaBuf = ByteBuffer.wrap(serialization.serialize(m2).get)
+          SerializedMeta(metaBuf, serManifest, serializer.identifier)
+        }
         // meta data, if any
         payload match {
           case SnapshotWithMetaData(_, m) =>
-            val m2 = m.asInstanceOf[AnyRef]
-            val serializer = serialization.findSerializerFor(m2)
-            val serManifest = Serializers.manifestFor(serializer, m2)
-            val metaBuf = ByteBuffer.wrap(serialization.serialize(m2).get)
-            Some(SerializedMeta(metaBuf, serManifest, serializer.identifier))
-          case _ => None
+            // FIXME can we remove support for the commercial version of this?
```

Comment:
May not matter but we give precedence the other way around for the events, first we check in the PR then we look at the payload.